core/src/main/java/com/redhat/lightblue/client/response/DefaultLightblueDataResponse.java

```diff
@@ -52,6 +52,25 @@ public int parseMatchCount() {
         return parseInt("matchCount");
     }
 
+    /**
+     * Returns a result metadata array where each element corresponds
+     * to the metadata for the result at the same index in processed
+     * array.
+     */
+    @Override
+    public ResultMetadata[] getResultMetadata() throws LightblueParseException {
+        JsonNode node=getJson().get("resultMetadata");
+        if(node instanceof ArrayNode) {
+            try {
+                return getMapper().readValue(node.traverse(),ResultMetadata[].class);
+            } catch (IOException e) {
+                throw new LightblueParseException("Error parsing lightblue response:"+getText()+"\n",e);
```

Comment:
This will only happen when this client is used against older Lightblue server version (from before update if current) - right?